include/drivers/bluetooth/hci_driver.h

```diff
@@ -128,6 +128,17 @@ struct bt_hci_driver {
 	 */
 	int (*open)(void);
 
+	/**
+	 * @brief Optional driver setup routine.
+	 *
+	 * This provides the driver the ability to have some vendor-
+	 * specific commands sent after the host has sent the initial
+	 * HCI_Reset, but before the host sends any other commands.
+	 *
+	 * @return 0 on success or negative error number on failure.
+	 */
+	int (*setup)(void);
```

Comment:
Since this is one of potentially multiple setup hooks I would be more descriptive in its name. Maybe `post_reset_setup()`?